packages/entity-server/src/routes/hierarchies/HierarchiesRoute.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request } from 'express';
+
+import { QueryConjunctions, Route } from '@tupaia/server-boilerplate';
+import { formatEntitiesForResponse } from './format';
+import { FlattenedHierarchy, HierarchyContext, HierarchyResponseObject } from './types';
+
+type ReqParams = Record<string, never>;
+type ResBody = HierarchyResponseObject[] | FlattenedHierarchy[];
+type ReqBody = Record<string, never>;
+
+interface ReqQuery {
+  fields?: string;
+  field?: string;
+}
+
+export interface HierarchyRequest extends Request<ReqParams, ResBody, ReqBody, ReqQuery> {
+  ctx: HierarchyContext;
+}
+
+export class HierarchyRoute extends Route<HierarchyRequest> {
+  public async buildResponse() {
+    const { field, fields } = this.req.ctx;
+
+    const allPermissionGroups = this.req.accessPolicy.getPermissionGroups();
+    const countryCodesByPermissionGroup: Record<string, string[]> = {};
+    // Generate lists of country codes we have access to per permission group
+    allPermissionGroups.forEach(pg => {
+      countryCodesByPermissionGroup[pg] = this.req.accessPolicy.getEntitiesAllowed(pg);
+    });
+
+    const entities = await this.req.models.entity.find(
+      {
+        type: 'project',
+        [QueryConjunctions.RAW]: {
+          // Pulls permission_group/country_code pairs from the project
+          // Returns any project where we have access to at least one of those pairs
+          sql: `(
+            SELECT COUNT(*) > 0 FROM
```

Comment:
Same logic as https://github.com/beyondessential/tupaia/blob/dev/packages/central-server/src/apiV2/GETProjects.js#L68

Reply:
Hmmm this code is pretty complex, it'd be nice to encapsulate it in one spot.

I'd suggest the `ProjectModel` in the `@tupaia/database` package? Some kind of `getAccessibleProjects(accessPolicy)` method. Then we can re-use in central-server and entity-server 👍

Alternatively, you could actually call out to the `central-api` to get the projects here? But I think I prefer that a little less...

Reply:
This is done now. However, I didn't refactor the mentioned code in `central-server` as it would require re-testing etc. I'd suggest for it to be done in a separate PR